src/generic_factory.h

```diff
@@ -121,28 +120,17 @@ class generic_factory
         DynamicDataLoader::deferred_json deferred;
 
     protected:
-        std::vector<T> list;
-        std::unordered_map<string_id<T>, int_id<T>> map;
-        std::unordered_map<std::string, T> abstracts;
+        std::map<std::string, T> objects;
+        std::map<std::string, T> abstracts;
+
+        std::map<std::string, std::set<std::string>> aliases;
+
+        std::unordered_map<int_id<T>, T *> map;
 
         std::string type_name;
         std::string id_member_name;
         std::string alias_member_name;
 
-        bool find_id( const string_id<T> &id, int_id<T> &result ) const {
-            result = id.get_cid();
-            if( is_valid( result ) && list[result].id == id ) {
-                return true;
```

Comment:
Note the three lines. When you do `string_id<T>::obj()`, it only does bounds check and returns an object by offset (using an index cached previously). That's very fast. The checks are even optional: their only purpose is safety in case of reloading. I was going to replace them with an event and drop entirely.